components/txn_types/src/lock.rs

```diff
@@ -28,12 +27,12 @@ const TXN_SIZE_PREFIX: u8 = b't';
 const MIN_COMMIT_TS_PREFIX: u8 = b'c';
 
 impl LockType {
-    pub fn from_mutation(mutation: &Mutation) -> LockType {
+    pub fn from_mutation(mutation: &Mutation) -> Option<LockType> {
         match *mutation {
-            Mutation::Put(_) | Mutation::Insert(_) => LockType::Put,
-            Mutation::Delete(_) => LockType::Delete,
-            Mutation::Lock(_) => LockType::Lock,
-            Mutation::CheckNotExists(_) => LockType::Noop,
+            Mutation::Put(_) | Mutation::Insert(_) => Some(LockType::Put),
+            Mutation::Delete(_) => Some(LockType::Delete),
+            Mutation::Lock(_) => Some(LockType::Lock),
+            Mutation::CheckNotExists(_) => None,
```

Comment:
@youjiali1995 Hi, I had remove Noop code, but it seems `from_mutation` need be called before https://github.com/tikv/tikv/pull/6715/commits/0568c72c31506f5d330bdc893b6bfbf815f4caa8#diff-10ac7ebd80ca4da7cccc8e3539f12eb4R357, so we still need to return a value here, but I'm not sure is this the right way in rust?(:smile: it's the first time I write rust)

Reply:
It's ok. I'm trying to find a better solution...